joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/passes/BasePassForAstProcessing.scala

```diff
@@ -0,0 +1,37 @@
+package io.joern.gosrc2cpg.passes
+
+import better.files.File
+import io.joern.gosrc2cpg.Config
+import io.joern.gosrc2cpg.astcreation.AstCreator
+import io.joern.gosrc2cpg.datastructures.GoGlobal
+import io.joern.gosrc2cpg.model.GoModHelper
+import io.joern.x2cpg.SourceFiles
+import io.shiftleft.codepropertygraph.generated.Cpg
+import io.shiftleft.passes.ConcurrentWriterCpgPass
+import org.slf4j.{Logger, LoggerFactory}
+
+abstract class BasePassForAstProcessing(
+  cpg: Cpg,
+  astFiles: List[String],
+  config: Config,
+  goMod: GoModHelper,
+  goGlobal: GoGlobal,
+  tmpDir: File
+) extends ConcurrentWriterCpgPass[String](cpg) {
```

Comment:
Note this PR: https://github.com/joernio/joern/pull/4689
```suggestion
) extends ForkJoinParallelCpgPass[String](cpg) {
```

Reply:
@DavidBakerEffendi is there any way I can load test the new flatgraph db version along with 
`ForkJoinParallelCpgPass` ?

Reply:
made the changes

Reply:
@pandurangpatil One would need to follow a similar chain of local builds as this PR: https://github.com/joernio/joern/pull/4693

Reply:
Just a note: no need to adapt your changes here, I'm taking care to reapply them while rebasing. 

To make it simpler for you, you can just pretend that there's not flatgraph branch. But either way, pre-applying those changes didn't hurt either :)